packages/core/js-client/src/jsServiceHost/JsServiceHost.ts

```diff
@@ -13,99 +13,110 @@
-     */
-    hasService(serviceId: string): boolean {
-        return this.commonHandlers.has(serviceId) || this.particleScopeHandlers.has(serviceId);
-    }
-
-    /**
-     * Removes all handlers associated with the specified particle scope
-     * @param particleId Particle ID to remove handlers for
-     */
-    removeParticleScopeHandlers(particleId: string): void {
-        this.particleScopeHandlers.delete(particleId);
-    }
-
-    /**
-     * Find call service handler for specified particle
-     * @param serviceId Service ID as specified in `call` air instruction
-     * @param fnName Function name as specified in `call` air instruction
-     * @param particleId Particle ID
-     */
-    getHandler(serviceId: string, fnName: string, particleId: string): GenericCallServiceHandler | null {
-        const key = serviceFnKey(serviceId, fnName);
-        const psh = this.particleScopeHandlers.get(particleId);
-        let handler: GenericCallServiceHandler | undefined = undefined;
-
-        // we should prioritize handler for this particle if there is one
-        // if particle-scoped handler exist for this particle try getting handler there
-        if (psh !== undefined) {
-            handler = psh.get(key);
-        }
+import {
+  CallServiceData,
+  CallServiceResult,
+  GenericCallServiceHandler,
+  IJsServiceHost,
+} from "./interfaces.js";
 
-        // then try to find a common handler for all particles with this service-fn key
-        // if there is no particle-specific handler, get one from common map
-        if (handler === undefined) {
-            handler = this.commonHandlers.get(key);
-        }
-
-        return handler || null;
+export class JsServiceHost implements IJsServiceHost {
+  private particleScopeHandlers = new Map<
+    string,
+    Map<string, GenericCallServiceHandler>
+  >();
+  private commonHandlers = new Map<string, GenericCallServiceHandler>();
+
+  /**
+   * Returns true if any handler for the specified serviceId is registered
+   */
+  hasService(serviceId: string): boolean {
+    return (
+      this.commonHandlers.has(serviceId) ||
+      this.particleScopeHandlers.has(serviceId)
+    );
+  }
+
+  /**
+   * Removes all handlers associated with the specified particle scope
+   * @param particleId Particle ID to remove handlers for
+   */
+  removeParticleScopeHandlers(particleId: string): void {
+    this.particleScopeHandlers.delete(particleId);
+  }
+
+  /**
+   * Find call service handler for specified particle
+   * @param serviceId Service ID as specified in `call` air instruction
+   * @param fnName Function name as specified in `call` air instruction
+   * @param particleId Particle ID
+   */
+  getHandler(
+    serviceId: string,
+    fnName: string,
+    particleId: string,
+  ): GenericCallServiceHandler | null {
+    const key = serviceFnKey(serviceId, fnName);
+    return this.particleScopeHandlers.get(particleId)?.get(key) ?? this.commonHandlers.get(key) ?? null;
```

Comment:
I don't know if you made CI autofix bot to work, but for sure your IDE doesn't format (if you did this change in your IDE of course) because this line is clearly too long and should'v been auto formatted by prettier to be shorter

do you plan bot autofix for a separate PR?

Reply:
I think yes, lets add it in separate PR.
https://linear.app/fluence/issue/DXJ-499/add-ci-autofix-bot-to-js-client